normalize.go

```diff
@@ -207,12 +216,19 @@ func (e *Error) NotEqual(err error) bool {
 
 // RedactErrorArg redacts the args by position if RedactLogEnabled is enabled.
 func RedactErrorArg(args []interface{}, position []int) {
-	if RedactLogEnabled.Load() {
+	switch RedactLogEnabled.Load() {
+	case RedactLogEnable:
 		for _, pos := range position {
 			if len(args) > pos {
 				args[pos] = "?"
 			}
 		}
+	case RedactLogMarker:
+		for _, pos := range position {
+			if len(args) > pos {
+				args[pos] = &redactFormatter{args[pos]}
```

Comment:
Previously the value was redacted with a question mark. I am having trouble following this- it seems like it is just putting brackets around the value rather than redacting. A test case might make this more clear.

Reply:
It seems like it could be error prone to do a find and replace unless you are using a character that is guaranteed to be escaped.
In a structured form redaction would be a lot more reliable.
I am switching to using structured values tagged using slog primitives which can be easily output with slog but also can be output to other loggers in structured form.
Perhaps redacted variables could be grouped together with a common prefix as an slog group.

* https://github.com/gregwebs/errors/blob/main/structured.go
* https://github.com/gregwebs/errors/blob/main/structured_test.go

Reply:
> It seems like it could be error prone to do a find and replace unless you are using a character that is guaranteed to be escaped.

And we adopt `‹`, this is not ascii brackets, but a special unicode "less than" character. This design spec is basically same as CRDB: https://github.com/cockroachdb/cockroach/blob/master/docs/RFCS/20200427_log_file_redaction.md.

>  In a structured form redaction would be a lot more reliable. I am switching to using structured values tagged using slog primitives which can be easily output with slog but also can be output to other loggers in structured form. Perhaps redacted variables could be grouped together with a common prefix as an slog group.

I prefer structured log, too. But I think that TiDB log format is just not structured. And a special redact format is somewhat difficult for users: there are other redact usages other than `pingcap/errors`. We'll need some more specific formats for this.

Reply:
That makes sense. I thought TiDB had adopted a uniform logging format with a uniform key-value syntax- that it is structured. It is a lot of work though to upgrade existing error creation to output structured information and it might perform slightly slower.